### DISCUSSION TITLE
Require Windows 10 or higher

### DISCUSSION THREAD

**Discussion by Seamus3131** [wixtoolset/issues#8051] [Questions]
Hi,

Using Wix v4 under VS2022

I try to make a condition to ensure OS is Windows 10 or higher

`
<Launch Condition="NOT (VersionNT >= v6.3)" Message="This application requires Windows 10 or higher." />
`

TBH I tried
`
<Launch Condition="VersionNT64 &lt; 603" Message="This application requires Windows 10 or higher." />
`
and many others, but on Windows 10 it still says "This application requires Windows 10 or higher."

Can you tell what I do wrong ?

Thanks

**Comment by ds-andreasor** [comment]
This seems you want to block an msi.
Nice people have written down the answer: https://www.firegiant.com/docs/wix/tutorial/sprint4/

**Reply by Seamus3131** [reply]
Thank you for the answer.
I got a look to the tutorial before writing the question.
They use MsiNTProductType and make a diff to 1 to know if it is a "modern" OS.

But for what I want to do the VersionNT64 seems to be the more accurate and making the condition in the same "shape" as the one it is done on the tutorial does not give the behaviour expected.

**Comment by chrpai** [comment]
A few more things to consider......

Why?  Is it a physical requirement or is it a support policy?
 
If physical, I try to detect and block on that instead.

If support policy, which version of windows 10?  There's been more a few now.

Don't forget to add the old `or Installed`  to your condition to make sure you somehow don't block uninstall.

I think I've forgotten more then I remember about the various appcompat hacks that Windows does so make sure whatever source of truth you use isn't lying to you. HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion might be a good place to look.

**Reply by Seamus3131** [reply]
Thank you for the answer.

It is for support policy, all OS that is not > Windows 10 will not be handled by support team. Number 1809 is minimum.
Thank you also for "or Installed" and for the registry key.

But I still would like to understand why the Condition is not working TBH.
Do you known (or someone knows) if there is a way to reflect the value of a macro inside the displayed message.
I want to see the values of some wix built-in variables : https://wixtoolset.org/docs/tools/burn/builtin-variables/ where some seems to be exactly what I look for.

for example something like 
<Launch Condition="NOT (VersionNT64 >= v6.3)" Message="This application requires Windows 10 or higher. $(sys.VersionNT64)" />
OR
<Launch Condition="NOT (VersionNT64 >= v6.3)" Message="This application requires Windows 10 or higher. [%sys.VersionNT64]" />
OR
<Launch Condition="NOT (VersionNT64 >= v6.3)" Message="This application requires Windows 10 or higher. ???" />

Thanks a lot
I saw there is a possible usage of 

Thanks a lot

**Comment by ds-andreasor** [comment]
But the `<Launch Condition>` element is for packages (.msi).
The tutorial is for packages as well.

The burn builtin variables are for bundle use (.exe) only.
And then a condition is an attribute of the bundle element. `<Bundle Condition="VersionNT64 &gt;= v10.0">`
